test/integration/connectors/test_kafka.py

```diff
@@ -98,6 +121,17 @@ def test_kafak_source_local_precheck_fail():
         indexer.precheck()
 
 
+@pytest.mark.tags(CONNECTOR_TYPE, SOURCE_TAG)
+def test_kafka_source_local_precheck_fail_no_topic(kafka_seed_topic: str):
+    connection_config = LocalKafkaConnectionConfig(bootstrap_server="localhost", port=29092)
+    indexer = LocalKafkaIndexer(
+        connection_config=connection_config,
+        index_config=LocalKafkaIndexerConfig(topic="topic", num_messages_to_consume=5),
+    )
+    with pytest.raises(SourceConnectionError):
+        indexer.precheck()
```

Comment:
Nitpick, it would be good to check that there is the "expected topic topic not detected in cluster" in the raised error, to make sure it's raising the correct SourceConnectionError.